fastly-edge-assignments/src/main.rs

```diff
@@ -0,0 +1,13 @@
+mod handlers;
+
+use fastly::http::{Method, StatusCode};
+use fastly::{Error, Request, Response};
+
+#[fastly::main]
+fn main(req: Request) -> Result<Response, Error> {
+    match (req.get_method(), req.get_path()) {
+        (&Method::POST, "/assignments") => handlers::handle_assignments(req),
+        (&Method::GET, "/health") => handlers::handle_health(req),
+        _ => Ok(Response::from_status(StatusCode::NOT_FOUND).with_body_text_plain("Not Found")),
+    }
+}
```

Comment:
I was hitting a CORS error running the SDK in the browser

<img width="657" alt="Screenshot 2024-11-22 at 12 01 52 AM" src="https://github.com/user-attachments/assets/42efa7a2-15b4-474b-8cc3-7aafbb2d0d02">

I think this is potentially a viable place to add CORS headers. If you agree, then consider adding them here for the preflight requests and responses. There are other places they could be configured though, e.g. terraform, so this is just a suggestion